File: backend/schemas/order.py
from pydantic import BaseModel
from typing import List, Optional
from datetime import datetime
from enum import Enum

class OrderStatus(str, Enum):
    PENDING = "ожидание"
    CONFIRMED = "подтверждено"
    PROCESSING = "обработка"
    SHIPPED = "отправлено"
    DELIVERED = "доставлено"
    CANCELLED = "отменено"

class OrderItemCreate(BaseModel):
    product_id: int
    quantity: int

class OrderBase(BaseModel):
    customer_id: int
    status: Optional[OrderStatus] = OrderStatus.PENDING

class OrderCreate(OrderBase):
    items: List[OrderItemCreate]

class OrderUpdate(BaseModel):
    status: Optional[OrderStatus] = None

class OrderItemResponse(BaseModel):
    id: int
    order_id: int
    product_id: int
    product_name: str = ""
    quantity: int
    price_per_unit: float
    total_price: float

    class Config:
        from_attributes = True

class OrderResponse(OrderBase):
    id: int
    total_amount: float
    created_at: datetime
    updated_at: datetime
    customer_name: str = ""
    org_name: str = ""
    items: List[OrderItemResponse] = []
    
    class Config:
        from_attributes = True

File: frontend/Models/Order.cs
namespace frontend.Models;
using System.Text.Json.Serialization;

public class Order
{
    public int Id { get; set; }
    
    [JsonPropertyName("customer_id")]
    public int CustomerId { get; set; }
    
    [JsonPropertyName("total_amount")]
    public decimal TotalAmount { get; set; }
    public string Status { get; set; } = string.Empty;
    
    [JsonPropertyName("created_at")]
    public DateTime CreatedAt { get; set; }
    
    [JsonPropertyName("updated_at")]
    public DateTime UpdatedAt { get; set; }
    [JsonPropertyName("customer_name")]
    public string CustomerName { get; set; } = string.Empty;

    [JsonPropertyName("items")]
    public List<OrderItem> Items { get; set; } = new();
}

public class OrderItem
{
    public int Id { get; set; }
    [JsonPropertyName("order_id")]
    public int OrderId { get; set; }
    [JsonPropertyName("product_id")]
    public int ProductId { get; set; }
    [JsonPropertyName("product_name")]
    public string ProductName { get; set; } = string.Empty;
    public int Quantity { get; set; }
    [JsonPropertyName("price_per_unit")]
    public decimal PricePerUnit { get; set; }
    [JsonPropertyName("total_price")]
    public decimal TotalPrice { get; set; }
}

public class OrderCreate
{
    [JsonPropertyName("customer_id")]
    public int CustomerId { get; set; }
    public List<OrderItemCreate> Items { get; set; } = new();
}

public class OrderItemCreate
{
    [JsonPropertyName("product_id")]
    public int ProductId { get; set; }
    public int Quantity { get; set; }
}

public class OrderUpdate
{
    public string? Status { get; set; }
}